usl_pipeline/cloud_functions/main.py

```diff
@@ -500,23 +515,56 @@ def _build_feature_matrix_from_archive(
       A tuple of (array, metadata) for the feature matrix and
       metadata describing the feature matrix.
     """
+    metadata: FeatureMetadata = FeatureMetadata()
+    elevation: Optional[geo_data.Elevation] = None
+    boundaries: Optional[list[Tuple[geometry.Polygon, int]]] = None
+    buildings: Optional[list[Tuple[geometry.Polygon, int]]] = None
+    green_areas: Optional[list[Tuple[geometry.Polygon, int]]] = None
+    soil_classes: Optional[list[Tuple[geometry.Polygon, int]]] = None
+    files_in_tar = []
+
     with tarfile.TarFile(fileobj=archive) as tar:
         for member in tar:
             fd = tar.extractfile(member)
             if fd is None:
                 continue
 
             name = pathlib.PurePosixPath(member.name).name
+            files_in_tar.append(name)
             # TODO: Group logic branch by path (per hazard model)
+            # Handle flood model input files (CityCat)
             if name == file_names.ELEVATION_TIF:
-                return _read_elevation_features(fd)
+                elevation, metadata = _read_elevation_features(fd)
+            elif name == file_names.BOUNDARIES_TXT:
+                boundaries = _read_polygons_from_byte_stream(fd)
+            elif name == file_names.BUILDINGS_TXT:
+                buildings = _read_polygons_from_byte_stream(fd)
+            elif name == file_names.GREEN_AREAS_TXT:
+                green_areas = _read_polygons_from_byte_stream(fd)
+            elif name == file_names.SOIL_CLASSES_TXT:
+                soil_classes = _read_polygons_from_byte_stream(fd)
             # Handle heat model input files (WPS)
             elif name.startswith("met_em") and name.endswith(".nc"):
                 return _build_wps_feature_matrix(fd)
             # TODO: handle additional archive members.
             else:
                 logging.warning(f"Unexpected member name: {name}")
 
+    if any((elevation, buildings, green_areas, soil_classes)):
+        if not all((elevation, buildings, green_areas, soil_classes)):
+            raise ValueError(
+                f"Some flood simulation data missing (see tar list: {files_in_tar})"
+            )
+        feature_matrix = feature_raster_transformers.transform_to_feature_raster_layers(
+            elevation,  # type: ignore
```

Comment:
would be good to leave a comment like
\# MyPy can't figure out that the all() call above prevents these from being None
to explain why we're ignoring these. We could also avoid having to ignore them by doing
```python
if elevation is None or buildings is None...
  raise ...
```
so MyPy will be able to tell these aren't None.

Reply:
Since it's 4 items that I would have to list in the `if` I prefer to just add a comment and ignore hints.